mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -193,11 +199,36 @@ class LinearRegression(override val uid: String)
     val intercept = if ($(fitIntercept)) yMean - dot(weights, Vectors.dense(featuresMean)) else 0.0
     if (handlePersistence) instances.unpersist()
 
+    val summary = generateTrainingResults(instances, lossHistory.result(), weights, intercept)
+
     // TODO: Converts to sparse format based on the storage, but may base on the scoring speed.
-    copyValues(new LinearRegressionModel(uid, weights.compressed, intercept))
+    copyValues(new LinearRegressionModel(uid, weights.compressed, intercept, summary))
+  }
+
+  private def generateTrainingResults(
```

Comment:
I'd prefer this method be in LinearRegressionTrainingResults, either as a constructor, or as a static factory method.

Also, you might want to pass in the LinearRegressionModel so that you can use its transform() method.  Rather than having LinearRegressionModel take summary in the constructor, it can start with it set to null; later on, you can set it using a setter method.